kernel/mempool.c

```diff
@@ -113,6 +113,41 @@ void *k_calloc(size_t nmemb, size_t size)
 	return ret;
 }
 
+void *k_realloc(void *ptr, size_t size)
+{
+	struct k_heap *heap, **heap_ref;
+	void *ret;
+
+	if (size == 0) {
+		k_free(ptr);
+		return NULL;
+	}
+	if (ptr == NULL) {
+		return k_malloc(size);
+	}
+	heap_ref = ptr;
+	ptr = --heap_ref;
+	heap = *heap_ref;
+
+	SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_heap_sys, k_realloc, heap, ptr);
```

Comment:
last comment:
Maybe this should be the start of this function.
But put here is ok for me.

Reply:
We don't have the reference to `heap` before that. It is at the same position as in `k_free()`